samplers/parser.go

```diff
@@ -19,9 +19,17 @@ type UDPMetric struct {
 	Value      interface{}
 	SampleRate float32
 	Tags       []string
-	LocalOnly  bool
+	Scope      MetricScope
 }
 
+type MetricScope int
```

Comment:
I think I mentioned this IRL, but one other approach is to use a bitmask, the way we're already doing for `Aggregates`. ie,

```go 
const (
 	Local  MetricScope = 1 << iota
 	Global    // this now magically has the value 0b10
 )

type UDPMetric struct {
 Scopes uint8
}
```

and then

```go

// make this mixed scope
m.Scopes = Local || Global

m2.Scopes = Local
```

...etc.


I think using bitmasks is probably cleaner, but I don't have super strong feelings about it.